app/models/ability.rb

```diff
@@ -60,16 +60,23 @@ def collection_abilities(user)
 
   def task_abilities(user)
     can %i[index create import_start import_confirm], Task
-
     alias_action :export_external_start, :export_external_check, :export_external_confirm, to: :export
-    can %i[crud export download add_to_collection], Task, user: {id: user.id}
+    can %i[show export download add_to_collection], Task do |task|
+      task.access_level_public? || task.in_same_group?(user) || task.author?(user)
+    end
+    can %i[update], Task do |task|
+      task.in_same_group_member?(user)
+    end
+    can %i[crud], Task do |task| # , user: {id: user.id}
+      task.author?(user) || task.in_same_group_admin?(user)
+    end
   end
 
   def task_file_abilities(user)
     can %i[download_attachment extract_text_data], TaskFile do |task_file|
       fileable = task_file.fileable
       task = fileable.is_a?(Task) ? fileable : fileable.task
-      task.can_access(user)
+      task.author?(user)
```

Comment:
according to #894 this should be possible for everyone who has access to the task itself

Reply:
Indeed. However, I am unsure about `extract_text_data`, since this behaves similar to a `update` operation (and changes the task / task file).

Reply:
You are right.
- `download_attachment` should be possible if you can view the task
- `extract_text_data` only if you can update the task